### DISCUSSION TITLE
How to create a new note using current selection?

### DISCUSSION THREAD

**Discussion by lokesh-krishna** [zk-org/zk#100] [Help] [answered]
I'm using Neovim 0.5 and native LSP. While I'm able to use all other functions, such as go to definition and auto-completion, I don't know how to go about creating a new note using the current selection, a feature that's mentioned in the editor integrations. 

Does this require any further setting up beyond just the basic lspconfig one?

**Comment by mickael-menu** [comment, marked as answer]
Yes this is actually implemented using LSP code actions. You need to bind a key to `vim.lsp.buf.range_code_action` as follows:

```lua
lspconfig.zk.setup({
  on_attach = function(client, bufnr)
    -- Key mappings
    local function buf_set_keymap(...) vim.api.nvim_buf_set_keymap(bufnr, ...) end
    local opts = { noremap=true, silent=false }
    
    buf_set_keymap("v", "<CR>", ":'<,'>lua vim.lsp.buf.range_code_action()<CR>", opts)
  end
})
```

Then, select a portion of text and hit Enter. You should see this:

<img width="681" alt="Screenshot 2021-11-11 at 20 26 54" src="https://user-images.githubusercontent.com/58686775/141356943-8b64c61c-cca6-45fb-97d8-d05434beea80.png">

**Reply by lokesh-krishna** [reply]
Thanks a lot :D
Could you also share what font that is in the screenshot you posted? Looks lovely!

**Reply by mickael-menu** [reply]
Sure, the font is Monaco (regular, with anti-aliasing) and the color scheme [Nord](https://github.com/arcticicestudio/nord-vim).
